src/std/misc/string.ss

```diff
@@ -185,3 +187,25 @@
 ;;  (string-whitespace? " \n\r \t") => #t
 (def (string-whitespace? s)
   (string-every char-whitespace? s))
+
+(def (random-word-char)
+  (declare (fixnum))
+  (def n (random-integer 63))
+  (integer->char
+   (+ n (cond
+	 ((< n 10) 48) ; 0-9
+	 ((< n 36) 55) ; A-Z
+	 ((< n 62) 61) ; a-z
+	 (else 33))))) ; _
+
+;; random-string returns a string consisting of
+;; regex word-boundary characters [a-zA-Z0-9_].
+(def (random-string (len 10))
+  (declare (fixnum))
+  (if (> len 0)
+    (call-with-output-string
```

Comment:
this could be a lot more efficient if we were creating a string of the right size and then setting the characters.
So something along these lines:
```
(let (str (make-string len))
  (do ((i 0 (1+ i))
        ((= i len))
    (string-set! str i (random-word-char)))
  str)
```

Reply:
Also, `(declare (not safe) (fixnum))` since there are no safety issues here (other than checking that len is a nonnegative fixnum.